common/lib/xmodule/xmodule/capa_base.py

```diff
@@ -632,6 +635,25 @@ def get_problem_html(self, encapsulate=True):
             'weight': self.weight,
         }
 
+        # Handle demand hints
+        demand_hints = self.lcp.tree.xpath("//problem/demandhint/hint")
+        show_hint_button = len(demand_hints) > 0
+        demand_hint = None  # if set to something, will show up in UI
+        if hint_index is None:
+            hint_index = 0  # bootstrap case, the client will start at 0
+        elif len(demand_hints) > 0:
+            _ = self.runtime.service(self, "i18n").ugettext  # pylint: disable=redefined-outer-name
+            hint_element = demand_hints[hint_index]
+            if len(demand_hints) == 1:
+                prefix = _('Hint: ')
+            else:
+                # Translators: e.g. "Hint 1 of 3" meaning we are showing the first of three hints.
+                # Do not translate hint_num and hints_count
+                prefix = _('Hint ({hint_num} of {hints_count}): ').format(hint_num=hint_index + 1,
```

Comment:
Hey, who doesn't have fond memories of correct i18n pluralization, so I see how that pattern jumped out. But in this case the ==1 is not pluralizing. It just means that if there is only one hint, we don't say "1 of 1" at the end, but if there's 2 or more we do say "1 of 2" or whatever as we show each hint. Each click of the hint button shows the next one.